.flake8

```diff
@@ -0,0 +1,15 @@
+[flake8]
+# black, the super-great auto-formatter has decided that 88 is a good number.
+# I'm inclined to agree, or not to dissagree.
+max-line-length = 88
+exclude =
+    test/*,
+ignore =
+    # H301: one import per line
+    H301,
+    # H306: imports not in alphabetical order
+    H306,
+    # W503:  line break before binary operator (this is no longer PEP8 compliant)
+    W503,
+    # E203: whitespace before ':' (this is not PEP8 compliant)
+    E203,
```

Comment:
It seems to be added by this PR, as it looks different from what exists in mbed-os-tools?
The rules look good to have, maybe we should acknowledge the issues and plan to fix them in the future, instead of ignoring the rules?

Reply:
Yes, I added these from mbed-tools. I've removed H301 and H306 as these do not need to be ignored. W503 is ignored as it goes against best practices, and E203 is ignored for PEP compliance, otherwise errors are raised due to formatting of list slicing.